sql/core/src/test/scala/org/apache/spark/sql/execution/HashAggregateCodegenInterruptionSuite.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution
+
+import scala.collection.mutable
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+import org.apache.logging.log4j.Level
+
+import org.apache.spark.{SparkConf, TaskEndReason, TaskKilled}
+import org.apache.spark.scheduler.{SparkListenerJobEnd, SparkListenerTaskEnd, SparkListenerTaskStart}
+import org.apache.spark.sql.QueryTest
+import org.apache.spark.sql.functions.sum
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.test.SharedSparkSession
+
+class HashAggregateCodegenInterruptionSuite extends QueryTest with SharedSparkSession {
+
+  override def sparkConf: SparkConf = {
+    super
+      .sparkConf
+      .set("spark.task.reaper.enabled", "true")
+      .set("spark.task.reaper.killTimeout", "10s")
+  }
+
+  test("SPARK-50806: HashAggregate codegen should be interrupted on task cancellation") {
+    import testImplicits._
+    withSQLConf(
+      SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key -> true.toString,
+      SQLConf.INTERRUPT_ON_CANCEL.key -> false.toString) {
+      var taskId = -1L
+      var isJobEnded = false
+      var taskEndReasons = new mutable.ArrayBuffer[TaskEndReason]
+      spark.sparkContext.addSparkListener(new org.apache.spark.scheduler.SparkListener {
+        override def onTaskStart(taskStart: SparkListenerTaskStart): Unit = {
+          taskId = taskStart.taskInfo.taskId
+        }
+
+        override def onTaskEnd(taskEnd: SparkListenerTaskEnd): Unit = {
+          taskEndReasons += taskEnd.reason
+        }
+
+        override def onJobEnd(jobEnd: SparkListenerJobEnd): Unit = {
+          isJobEnded = true
+        }
+      })
+
+      val logAppender = new LogAppender("")
+      withLogAppender(logAppender, level = Some(Level.INFO)) {
+        spark.sparkContext.setJobGroup("SPARK-50806", "SPARK-50806", false)
+        val slowDF = spark.range(1, 100000).rdd.mapPartitions { iter =>
```

Comment:
Could you add a comment that this test case assumption: this should be `100000`(1k) because we check it per 1k rows?

Reply:
Updated.